diff --git a/src/pixhawk_platform.cpp b/src/pixhawk_platform.cpp
index 2d0913c..891eda8 100644
--- a/src/pixhawk_platform.cpp
+++ b/src/pixhawk_platform.cpp
@@ -427,10 +427,6 @@ void PixhawkPlatform::resetTrajectorySetpoint()
 
 void PixhawkPlatform::resetAttitudeSetpoint()
 {
-  px4_attitude_setpoint_.pitch_body = NAN;
-  px4_attitude_setpoint_.roll_body = NAN;
-  px4_attitude_setpoint_.yaw_body = NAN;
-
   px4_attitude_setpoint_.q_d = std::array<float, 4>{0, 0, 0, 1};
   px4_attitude_setpoint_.thrust_body = std::array<float, 3>{0, 0, -min_thrust_};
 }
@@ -776,9 +772,9 @@ void PixhawkPlatform::px4GpsCallback(const px4_msgs::msg::SensorGps::SharedPtr m
   }
   nav_sat_fix_msg.status.service = sensor_msgs::msg::NavSatStatus::SERVICE_GPS;  // DEFAULT
 
-  nav_sat_fix_msg.latitude = msg->lat;
-  nav_sat_fix_msg.longitude = msg->lon;
-  nav_sat_fix_msg.altitude = msg->alt_ellipsoid;
+  nav_sat_fix_msg.latitude = msg->latitude_deg;
+  nav_sat_fix_msg.longitude = msg->longitude_deg;
+  nav_sat_fix_msg.altitude = msg->altitude_ellipsoid_m;
 
   if (!std::isnan(msg->eph) && !std::isnan(msg->epv)) {
     // Position uncertainty --> Diagonal known
@@ -793,10 +789,6 @@ void PixhawkPlatform::px4GpsCallback(const px4_msgs::msg::SensorGps::SharedPtr m
     nav_sat_fix_msg.position_covariance = {-1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0};
     nav_sat_fix_msg.position_covariance_type = sensor_msgs::msg::NavSatFix::COVARIANCE_TYPE_UNKNOWN;
   }
-  nav_sat_fix_msg.latitude = nav_sat_fix_msg.latitude / 1e7;
-  nav_sat_fix_msg.longitude = nav_sat_fix_msg.longitude / 1e7;
-  nav_sat_fix_msg.altitude = nav_sat_fix_msg.altitude / 1e3;
-
   gps_sensor_ptr_->updateData(nav_sat_fix_msg);
 }
 
@@ -811,8 +803,8 @@ void PixhawkPlatform::px4BatteryCallback(const px4_msgs::msg::BatteryStatus::Sha
   battery_msg.temperature = msg->temperature;
   battery_msg.current = msg->current_a;
   battery_msg.charge = NAN;
-  battery_msg.capacity = msg->capacity;
-  battery_msg.design_capacity = msg->design_capacity;
+  battery_msg.capacity = msg->capacity / 1000.0f;
+  battery_msg.design_capacity = msg->capacity / 1000.0f;
   battery_msg.percentage = 100.0 * msg->remaining;
   // TODO(miferco97): config file with battery settings
   battery_msg.power_supply_status = sensor_msgs::msg::BatteryState::POWER_SUPPLY_STATUS_UNKNOWN;
@@ -824,7 +816,7 @@ void PixhawkPlatform::px4BatteryCallback(const px4_msgs::msg::BatteryStatus::Sha
   battery_msg.cell_voltage = {};
   battery_msg.cell_temperature = {};
   battery_msg.location = '0';
-  battery_msg.serial_number = std::to_string(msg->serial_number);
+  battery_msg.serial_number = std::to_string(msg->id);
 
   if (msg->warning >= 0) {
     RCLCPP_WARN_ONCE(this->get_logger(), "Battery warning #%d", msg->warning);
